datadriven/src/sgpp/datadriven/datamining/modules/fitting/ModelFittingDensityDifferenceEstimationOnOff.cpp

```diff
@@ -184,6 +192,45 @@ void ModelFittingDensityDifferenceEstimationOnOff::update(DataMatrix& newDataset
   }
 }
 
+double ModelFittingDensityDifferenceEstimationOnOff::computeResidual(
+    DataMatrix& validationData) const {
+  DataVector bValidation = online->computeWeightedBFromBatch(
+      validationData, *grid, this->config->getDensityEstimationConfig(), true);
```

Comment:
@janschopohl shouldn't this be a computeWeightedTwoDatasetsBFromBatch? (or however the function is called to compute the P-Q rhs term....)